packages/runtime/src/runtime/modal/index.tsx

```diff
@@ -142,23 +163,24 @@ export const ModalWrapper: React.FC<PropsWithChildren> = ({ children }) => {
           if (oldModalState.length === 0) {
             return oldModalState;
           }
-          return [
+          const nextState = [
             ...oldModalState.slice(0, -1),
             {
               ...oldModalState[oldModalState.length - 1],
               visible: false,
             },
           ];
```

Comment:
@anserwaseem what was the purpose of this?

Reply:
it's there such that when a new modal is opened via `navigateModalScreen`, the previous one is hidden